### DISCUSSION TITLE
Subscriptions Listener Timeouting or not working some times or do not working.

### DISCUSSION THREAD

**Discussion by LeCyreaxYT** [pocketbase/pocketbase#3204] [Q&A]
``I use the Subscriptions to handle my sync with all Clients and its important thats all Synced.. Now i have the Problem why its running in the Problem with Timeouts or Erros
![image](https://github.com/pocketbase/pocketbase/assets/38440842/f3aca5be-dbfb-42da-9024-71c4ad29420e)

```typescript
this.$pocketbase.getCollection("streifen_members").subscribe("*", async(data: RecordSubscription < Record > ) = >{
  if (data.action == "create") {
    const patrolMember: Record = await this.$pocketbase.getCollection("streifen_members").getFirstListItem(`id = "${data.record.id}"`, {
      expand: "user.rank",
    })

    if (patrolMember) {
      this.patrolMembers.push(patrolMember);
    }
  } else if (data.action == "update") {
    const patrolMember: Record = await this.$pocketbase.getCollection("streifen_members").getFirstListItem(`id = "${data.record.id}"`, {
      expand: "user.rank",
    })

    if (patrolMember) {
      const index = this.patrolMembers.findIndex((x: any) = >x.id == data.record.id);
      if (index != -1) {
        this.patrolMembers[index] = patrolMember;
      }
    }
  } else if (data.action == "delete") {
    const index = this.patrolMembers.findIndex((x: any) = >x.id == data.record.id);
    if (index != -1) {
      this.patrolMembers.splice(index, 1);
    }
  }

  await this.refreshPatrolsMembers();
  this.syncDate = new Date().toISOString();
});

this.$pocketbase.getCollection("streifen").subscribe("*", async(data: RecordSubscription < Record > ) = >{
  if (data.action == "create") {
    const patrol = await this.$pocketbase.getCollection("streifen").getOne(data.record.id);

    if (patrol) {
      this.patrols.push(patrol);
    }
  } else if (data.action == "update") {
    const patrol = await this.$pocketbase.getCollection("streifen").getOne(data.record.id);

    if (patrol) {
      const index = this.patrols.findIndex((x: any) = >x.id == data.record.id);
      if (index != -1) {
        this.patrols[index] = patrol;
      }
    }
  } else if (data.action == "delete") {
    const index = this.patrols.findIndex((x: any) = >x.id == data.record.id);
    if (index != -1) {
      this.patrols.splice(index, 1);
    }
  }

  await this.refreshPatrolsMembers();
  this.syncDate = new Date().toISOString();
});
```

**Comment by ganigeorgiev** [comment]
There are not enough details to help you debug it or even understand what actually is the error in the above.

- What version of the JS SDK do you use?
  Note that the official JS SDK doesn't have `getCollection()` but `collection()` method? Do you use something else?

- Where do you run PocketBase - locally, fly.io, etc.? 
  _If deployed on fly.io, note that they have 60s connections limit, but for the end user this shouldn't matter because the client will reconnect automatically._

- How do you interact with PocketBase - client-side, server-side or both?
  If server-side, make sure that your node server (or any other JS runtime you use) is properly configured and allow long-living connections.

- Do you have a reverse proxy (nginx, apache, caddy, etc.)?
  If Yes, again check its configuration (you can find sample minimal config in https://pocketbase.io/docs/going-to-production/#using-reverse-proxy).

To rule out PocketBase or JS SDK issue, you can try to create a static index.html in `pb_public/index.html` and test it directly:

```html
<!DOCTYPE html />
<html>
    <head>
        <meta charset="utf-8" />
        <meta name="viewport" content="width=device-width, initial-scale=1" />
        <title>Test</title>
    </head>
    <body>
        <script type="module">
            import PocketBase from "https://cdn.jsdelivr.net/gh/pocketbase/js-sdk@master/dist/pocketbase.es.mjs";

            const pb = new PocketBase("http://127.0.0.1:8090");

            pb.collection("streifen_members").subscribe("*", (e) => {
                console.log(e)
            })
        </script>
    </body>
</html>
```
(_open `http://127.0.0.1:8090` in your browser and check the console for messages_)

If none of the above helps, please create a **minimal** reproducible repo and I'll have a look.
